packages/firebase_auth/lib/firebase_auth.dart

```diff
@@ -192,4 +192,28 @@ class FirebaseAuth {
     final FirebaseUser currentUser = new FirebaseUser._(data);
     return currentUser;
   }
+
+  /// Links google account with current user and returns [Future<FirebaseUser>]
+  ///
+  /// throws [PlatformException] when
```

Comment:
A `PlatformException` is super generic and in general should be reserved for plugin-unspecific errors. Can you throw a more platform-specific exception in these cases?

Reply:
> Basically, in my opinion the course would be to provide a set of Firebase Auth Exceptions for all Firebase Auth.

Yeah, that would probably be the right thing to do here (and updating the existing methods to use those). Would you be up for doing that refactoring?

Reply:
Well, for sure I wouldn't be able to do it all by myself since I have no way to develop iOS plugin.
Also I have found this issue flutter/flutter#11111 which also considering refactoring firebase, not sure how strongly it is connected to our problem.
I can create an issue and try to describe is it as detailed as I can but I am not confident enough to say that I am going to do it <b>now</b> especially with my current knowledge. I hope you understand.

Reply:
Makes sense. Since this is a pre-existing condition, can you add a TODO to the code that this should be changed to throw custom exceptions?

Reply:
I added TODO in dart class. Is this what you meant?